### ISSUE TITLE
mouse cursor & coordinates in X window apps (canvas) on none primary display monitors seems inconsistent for multiple display monitors machine

### ISSUE THREAD

**Issue by quantummixer** [microsoft/wslg#288]
# Environment

```none
Windows build number: 10.0.21390.1010
Your Distribution version: Ubuntu Release: 18.04
Your WSLg version: Window Subsystem for Linux WSLg Preview version 1.0.22
```

# Steps to reproduce

**WSL logs**: 

* Attach WSLg logs from  `/mnt/wslg`

You can access the wslg logs using explorer at: `\\wsl$\<Distro-Name>\mnt\wslg` (e.g.: `\\wsl$\Ubuntu-20.04\mnt\wslg`)

* [pulseaudio.log](https://github.com/microsoft/wslg/files/6644641/pulseaudio.log)
* [weston.log](https://github.com/microsoft/wslg/files/6644642/weston.log)
* [versions.txt](https://github.com/microsoft/wslg/files/6644640/versions.txt)

#  Expected behavior

As a new user of `wslg`, I tried testing the `gimp` app from the Ubuntu 18.04 wsl install.  I have 3 monitors connected to my laptop with different orientation and pixel resolutions:  primary (landscape), second (portrait orientation), third (landscape).  

When I move the `gimp` window away from my primary display to other working displays, all mouse interactions to that `gimp` window should continue to work flawlessly.  

# Actual behavior

It seems the mouse cursor is "lost" or not "rendered" when it is within the `gimp` GUI window.   When I used the mouse cursor to enter the bounding box of the `gimp` GUI window, I tried to click on application GUI buttons but the location seems to be off.  

I fired up another GUI application `Microsoft Edge browser for Linux` installed from the same `Ubuntu 18.04` which loads up without any issue on my primary display and all of my mouse cursor interaction with any part of the window is accurate and precise.  

I then proceed to open a new window to a webpage, https://isc.tamu.edu/~lewing/linux/ and move the new `Edge` window to the other display window, similar odd behavior follows.  Again some of the mouse cursor interaction with the rendered web page seems to be working, but when the mouse cursor is moved to `Edge functional bar` or `ribbon`, the region above the rendered webpage, the mouse cursor disappears again.  The mouse cursor offset is off or causing teleportation effects when I tried to move in and out of the window in my none primary display with attempts to click the `Edge functional UI buttons`.

**Comment by onomatopellan** [comment]
Did you try installing WSLg version 1.0.23? https://github.com/microsoft/wslg/releases/tag/v1.0.23

**Comment by quantummixer** [comment]
Thank you for the quick reply!  

I followed the upgrade instruction written here:   https://github.com/microsoft/wslg#updating-wsl--wslg

I opened an Administrator: Windows PowerShell:
```
PS C:\WINDOWS\system32> wsl --update
Checking for updates...
No updates are available.
Kernel version: 5.10.16
```

I doubled checked the `Apps & features` screen and queried `Windows Subsytsem for Linux WSLg Preview`, still showing `1.0.22` with a date of `6/8/2021`.  

It seems my attempt to update just the WSLg to `1.0.23`  from `1.0.22` doesn't work.  

I performed additional command to get more status and contextual information:
```
PS C:\WINDOWS\system32> wsl --status
Default Distribution: Ubuntu
Default Version: 2

Windows Subsystem for Linux was last updated on 6/8/2021
WSL automatic updates are on.

Kernel version: 5.10.16
```

Is there another command option example you can provide that can help me to force update WSLg to update from `1.0.22` to `1.0.23` when WSL kernel is up-to-date?

Thank you all for this great community and great product!

**Comment by onomatopellan** [comment]
Just download the msi file it from here https://github.com/microsoft/wslg/releases/download/v1.0.23/wsl_graphics_support_x64.msi and install it. In Add/Remove programs you can see the version WSLG you have installed.

**Comment by quantummixer** [comment]
Thank you so much!  Your suggestion worked!  I have been trying to follow the instructions word by word to make sure I didn't mess up anything in the sequence of installs / updates.  

I have confirmed that version `1.0.23` does contain the [patch](https://github.com/microsoft/weston-mirror/commit/40ea4649) that fixed this issue I have reported.   Please close this issue out as resolved by version `1.0.23`.  

Thank you so much for you help!
